app/controllers/concerns/user_authentication.rb

```diff
@@ -10,14 +10,45 @@ module UserAuthentication
   private
 
   def authenticate_user!
-    redirect_to :login if session[:user_id].blank?
+    logger.debug "authenticate_user! About to call warden.authenticate!"
+    warden.authenticate!
   end
 
   def current_user
-    @current_user ||= User.find_by!(members_user_id: session[:user_id])
+    @current_user ||= begin
+      logger.debug "authenticate_user About to call warden.authenticate"
+      warden.authenticate
+    end
   end
 
   def user_logged_in?
-    !current_user.nil?
+    !!current_user
+  end
+
+  # def user_session
+  #   current_user && warden.session(:user)
+  # end
+
+  def warden
+    request.env['warden'].tap do |w|
+      logger.debug "Warden: #{w}"
+    end
   end
 end
```

Comment:
Remove commented out code

Reply:
Decided to leave some of it in as a reminder of how Devise does it. For future refactoring/enhancement.